cmake/modules/ddr/DDRSetStub.cmake.in

```diff
@@ -0,0 +1,181 @@
+
+		get_filename_component(abs_file "${source_file}" ABSOLUTE BASE_DIR "${OPT_SOURCE_DIR}")
+		file(RELATIVE_PATH output_file "${OPT_SOURCE_DIR}" "${abs_file}")
+		string(REGEX REPLACE "\\.\\.(/|$)" "__\\1" output_file "${output_file}")
+		set(output_file "${OPT_TARGET}/${output_file}")
+		get_filename_component(output_dir "${output_file}" DIRECTORY)
+		file(MAKE_DIRECTORY "${output_dir}")
+
+		add_filename_extension(stub_file "${output_file}" ".stub")
+		add_filename_extension(annt_file "${output_file}" ".annt")
+
+		list(APPEND stub_files "${stub_file}")
+		list(APPEND annotated_files "${annt_file}")
+
+		add_custom_command(
+			OUTPUT "${stub_file}"
+			DEPENDS
+				${abs_file}
+				${DDR_SUPPORT_DIR}/cmake_ddr.awk
+				${DDR_SUPPORT_DIR}/GenerateStub.cmake
+				COMMAND ${CMAKE_COMMAND} -DAWK_SCRIPT=${DDR_SUPPORT_DIR}/cmake_ddr.awk -Dinput_file=${abs_file} -Doutput_file=${stub_file} -P ${DDR_SUPPORT_DIR}/GenerateStub.cmake
+		)
+
+		#TODO dont hardcode command
+		set(pp_command "gcc")
+		list(APPEND pp_command ${BASE_ARGS} "-E" "${stub_file}" )
+		add_custom_command(
+			OUTPUT "${annt_file}"
+			DEPENDS "${stub_file}"
+			COMMAND  ${pp_command} | awk "/^\$/{next} /^DDRFILE_BEGIN /,/^DDRFILE_END /{print \"@\" $0}"  > ${annt_file}
+			VERBATIM
+		)
+	endforeach()
+
+	if(OPT_OUTPUT_STUBS)
+		set("${OPT_OUTPUT_STUBS}" "${stub_files}" PARENT_SCOPE)
+	endif()
+
+	if(OPT_OUTPUT_ANNOTATED)
+		message(STATUS "setting annotated file for ${target} ${annotated_files}")
+		set("${OPT_OUTPUT_ANNOTATED}" "${annotated_files}" PARENT_SCOPE)
+	endif()
+endfunction(process_source_files)
+
+set(stubannotated_files )
+set(target_files )
+
+function(process_target target)
+	file(MAKE_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}/${target}")
+	file(STRINGS "${CMAKE_SOURCE_DIR}/${target}.txt" target_config)
+
+	cmake_parse_arguments("OPT" "" "SOURCE_DIR;OUTPUT_FILE" "INCLUDE_PATH;DEFINES;SOURCES;HEADERS;PREINCLUDES" ${target_config})
+
+	if(OPT_OUTPUT_FILE)
+		set(target_files ${target_files} "${OPT_OUTPUT_FILE}")
+	endif()
+
+
+	process_source_files(
+		${OPT_SOURCES}
+		TARGET "${target}"
+		SOURCE_DIR "${OPT_SOURCE_DIR}"
+		OUTPUT_ANNOTATED project_annt_src
+		INCLUDE_DIRS ${OPT_INCLUDE_PATH}
+
+	)
+	list(APPEND annotated_files ${project_annt_src})
+
+	# The only difference between source files and header files, is that header files may need
+	# other headers pre-included so that preprocessor macros work properly
+	process_source_files(
+		${OPT_HEADERS}
+		TARGET "${target}"
+		SOURCE_DIR "${OPT_SOURCE_DIR}"
+		OUTPUT_ANNOTATED project_annt_hdr
+		#TODO need to add pre-include files
+	)
+	list(APPEND annotated_files ${project_annt_hdr})
+
+	# Bump changes up to parent scope
+	set(annotated_files "${annotated_files}" PARENT_SCOPE)
+
+endfunction(process_target)
+
+foreach(target IN LISTS DDR_INPUTS)
+	process_target("${target}")
+endforeach()
+
+
+#add_custom_target(annotated DEPENDS ${annotated_files} )
+
+
+add_custom_command(
+    OUTPUT macro_list
```

Comment:
Why not use the default name 'macroList' so downstream projects don't have have to change?